omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -11,59 +12,30 @@ const { v4: uuidv4 } = require('uuid');
  * @param {JSON} req contains request headers and payload businessEmail,fullName,password,photoUrl,provider
  */
 exports.register = async function (req, res) {
-    const body = req.body;
-    await User.findUser(body.businessEmail, function (err, data) {
-        if (data && data.userId) {
-            res.status(400).json({
-                success: false,
-                message: 'Email Already Exists',
-            });
-        }
-        if (err) {
-            res.status(400).json({
-                success: false,
-                message: 'Something went wrong while fetching user!'
-            });
-        }
-    });
-    const salt = await bcrypt.genSalt(10);
-    const password = await bcrypt.hash(body.password, salt);
-    delete body.password; 
-    delete body.confirmPassword;
-    const user = {
-        ...body,
-        userId: uuidv4(),
-    }
     try {
-        await User.registerUser({ ...user,password }, function (err, data) {
+        const body = req.body;
+        const [registerUser] = await User.findUser(body.businessEmail);
+        if (registerUser && registerUser.userId) {
+            return res.status(400).json({ success: false, message: 'Registration failed, Email already exists!' });
+        }
+        const salt = await bcrypt.genSalt(10);
+        const password = await bcrypt.hash(body.password, salt);
+
+        // deleting password and confirm password as both are not required to store
+        delete body.password;
+        delete body.confirmPassword;
+
+        const user = { ...body, userId: uuidv4() };
+        await User.registerUser({ ...user, password });
+        jwt.sign(user, process.env.JWT_KEY, { expiresIn: process.env.EXPIRE_IN }, function (err, token) {
             if (err) {
-                res.json({
-                    success: false,
-                    message: 'Something went wrong while registering new user!'
-                });
-            }
-            if (data) {
-                jwt.sign(user, process.env.JWT_SECRET, { expiresIn: process.env.EXPIRE_IN }, function (err, token) {
-                    if (err) {
-                        res.status(400).json({
-                            success: false,
-                            message: 'Something went wrong while registering new user!'
-                        });
-                    }
-                    res.json({
-                        success: true,
-                        message: 'Registered successfully!',
-                        userInfo: user,
-                        authToken: token
-                    });
-                });
+                return res.status(400).json({ success: false, message: 'Something went wrong while registering new user!' });
```

Comment:
Codes in the `400` range indicate that the client made an error. Prefer codes from the `500` range for unexpected errors without a clear cause.

Reply:
the above issue has been resolved.